src/UI/Implementation/Component/Input/Field/Duration.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Input\Field;
+
+use ILIAS\UI\Component as C;
+use ILIAS\Data\Factory as DataFactory;
+use ILIAS\Transformation\Factory as TransformationFactory;
+use ILIAS\Validation\Factory as ValidationFactory;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+
+
+/**
+ * This implements the duration input group.
+ */
+class Duration extends Group implements C\Input\Field\Duration {
+
+	use ComponentHelper;
+
+	/**
+	 * @var string
+	 */
+	protected $format = "MM/DD/YYYY";
+
+	/**
+	 * @var \DateTime
+	 */
+	protected $min_date;
+
+	/**
+	 * @var \DateTime
+	 */
+	protected $max_date;
+
+	/**
+	 * @var bool
+	 */
+	protected $use_time_glyph = false;
+
+
+	public function __construct(
+		DataFactory $data_factory,
+		ValidationFactory $validation_factory,
+		TransformationFactory $transformation_factory,
+		Factory $field_factory,
+		$label,
+		$byline
+	) {
+
+		$inputs = [
+			$field_factory->date('start'),
+			$field_factory->date('end')
+		];
+
+		parent::__construct($data_factory, $validation_factory, $transformation_factory, $inputs, $label, $byline);
+		$this->addTransformation($transformation_factory);
+		$this->addValidation($validation_factory);
+		$this->validation_factory = $validation_factory;
+	}
+
+
+	protected function addTransformation(TransformationFactory $transformation_factory) {
+		$duration = $transformation_factory->custom(function($v) {
+			list($from, $until) = $v;
+			if($from && $until) {
+				return ['start'=>$from, 'end'=>$until, 'interval'=>$from->diff($until)];
+			}
+			return '';
```

Comment:
This is odd. So the field either contains a `\DateDuration` or an empty string? Please throw here if input is not what you expected.